cmake/utils.cmake

```diff
@@ -160,55 +157,21 @@ endfunction()
 
 #
 # mayaUsd_promoteHeaderList(
-#                        [SUBDIR <optional sub-directory>])
-#                        [FILES <list of files>]
+#                        [SUBDIR  <sub-directory name>]
+#                        [FILES   <list of files>]
+#                        [BASEDIR <sub-directory name>])
+#
+#   SUBDIR     - sub-directory in which to promote files.
+#   FILES      - list of files to promote.
+#   BASESDIR   - base dirctory where promoted headers are installed into.
+#                if not defined, mayaUsd subdirectory is used by default. 
 #
-#   SUBDIR        - optional sub-directory in which to promote files.
-#   FILES         - list of files to promote.
```

Comment:
removed duplicated `mayaUsd_promoteHeaderListWithSubdir` and consolidated some of it's functionalities into `mayaUsd_promoteHeaderList`